test/reline/yamatanooroti/termination_checker.rb

```diff
@@ -19,8 +19,8 @@ def terminated?(code)
 
 class AutoIndent < RubyLex
   def initialize
-    set_input(self)
-    context = Struct.new(:auto_indent_mode, :workspace).new(true, nil)
+    set_input(self, context: nil)
+    context = Struct.new(:auto_indent_mode, :workspace, :local_variables).new(true, nil, [])
```

Comment:
Just curious: does passing this context to `set_input` work?

Reply:
The [only `set_input` invocation](https://github.com/ruby/irb/blob/d30c0c9943514a7c24500411838fb060df49b6c1/lib/irb.rb#L546-L564) in IRB expects `context` to always be present (otherwise `@context.io` would fail). And you can see that `set_auto_indent` is then called with the same context right after it.
So if these tests try to simulate how IRB works, I think `set_input` should have `context` too.

Reply:
I have the same opinion with @st0012.

I can probably pass context with this. How does it look?

```diff
diff --git a/test/reline/yamatanooroti/termination_checker.rb b/test/reline/yamatanooroti/termination_checker.rb
index 665fd91..fda1fb3 100644
--- a/test/reline/yamatanooroti/termination_checker.rb
+++ b/test/reline/yamatanooroti/termination_checker.rb
@@ -1,5 +1,5 @@
 require 'ripper'
-require 'irb/ruby-lex'
+require 'irb'

 class TerminationChecker < RubyLex
   def terminated?(code)
@@ -19,7 +19,9 @@ end

 class AutoIndent < RubyLex
   def initialize
-    set_input(self, context: nil)
+    IRB.init_config(nil)
+    context = IRB::Context.new(nil, IRB::WorkSpace.new(Object.new), Object.new)
+    set_input(self, context: context)
     context = Struct.new(:auto_indent_mode, :workspace, :local_variables).new(true, nil, [])
     set_auto_indent(context)
   end
```

Reply:
I kinda feel this is an anti-pattern though. We're trying to simulate a downstream dependency in an upstream library's tests 🤔 

Maybe these tests actually belong to `IRB` instead of `reline`?

Reply:
I agree with Stan.
test_yamatanooroti looks like E2E test and might be injected some dependency that Reline basically should not be responsible for.
I guess Struct-mock is enough for now while it's not a perfect solution.
How about this way of naming the Struct?
👇

Reply:
LGTM 👍

Reply:
This seems better! LGTM 🌟